src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -737,7 +807,26 @@ private bool ValidatePackagesSha512(PackagesLockFile lockFile, LockFile assetsFi
                     _request.Project.RestoreMetadata.CacheFilePath = null;
                 }
             }
-            return (cacheFile, noOp);
+            return (cacheFile, noOp, cacheFileAge);
+
+            static bool CacheFileExists(string path, out TimeSpan? cacheFileAge)
```

Comment:
I think this local function has 2 responsibilities, one to check if the file exists and second one to fetch the cache file age. Can we have separate methods? I am okay with doing this in a follow-up PR.

Reply:
If we call one method to check if the file exists, and another to get the timestamp, then we're going to go to the kernel twice, which is bad for performance.